packages/runtime/src/webcontainer/on-demand-boot.ts

```diff
@@ -0,0 +1,50 @@
+/**
+ * Lightweight wrapper around WebContainer.boot
+ * to only boot if requested on iOS as typically
+ * boot might risk the page to run out of memory.
+ */
+import { WebContainer, type BootOptions } from '@webcontainer/api';
+import { withResolvers } from '../utils/promises.js';
+
+let blocked: undefined | boolean;
+const blockingStatus = withResolvers<void>();
+
+export async function safeBoot(options: BootOptions) {
+  if (typeof blocked === 'undefined') {
+    blocked = isRestricted();
+  }
+
+  if (blocked) {
+    await blockingStatus.promise;
```

Comment:
We now prevent WebContainer from booting on iOS to avoid crashing the page as it sometimes does. Instead the idea is to later let them proceeed with the boot by pressing a button.